src/wallet/rpcwallet.cpp

```diff
@@ -3470,6 +3470,30 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid zaddr.");
     }
 
+    // A non-unified address argument that is a receiver within a
+    // unified address known to this wallet is not allowed.
```

Comment:
Wait, why not? This seems like an odd exception. What if the provider of the bare address doesn't know the full UA? Why should an address that happens to be part of a UA known to the wallet behave differently to any other bare address? I would definitely have pushed back on this if I hadn't been distracted by unrelated stuff yesterday.